### PR TITLE
[fix][test] flaky AdminApi2Test.cleanup

### PR DESCRIPTION
**Pull request by poorbarcode**
Fixes: #16980
Fixes: #17530

- https://github.com/apache/pulsar/issues/16980
- https://github.com/apache/pulsar/issues/17530

### Motivation

When runs `AdminApi2Test.testTopicPoliciesWithMultiBroker` or `BrokerServiceLookupTest.testModularLoadManagerSplitBundle`, you can see some logs like this:

```
2022-09-27T22:21:13,414 - WARN  - [metadata-store-12286-1:LeaderElectionImpl@310] - Leadership released for /loadbalance/leader
2022-09-27T22:21:13,418 - INFO  - [metadata-store-12319-1:LeaderElectionImpl@183] - Acquired leadership on /loadbalance/leader
2022-09-27T22:21:13,418 - INFO  - [metadata-store-12319-1:PulsarService@1067] - This broker was elected leader
2022-09-27T22:21:13,418 - WARN  - [metadata-store-12319-1:LeaderElectionImpl@310] - Leadership released for /loadbalance/leader
2022-09-27T22:21:13,422 - INFO  - [metadata-store-12286-1:LeaderElectionImpl@183] - Acquired leadership on /loadbalance/leader
2022-09-27T22:21:13,422 - INFO  - [metadata-store-12286-1:PulsarService@1067] - This broker was elected leader
2022-09-27T22:21:13,423 - WARN  - [metadata-store-12286-1:LeaderElectionImpl@310] - Leadership released for /loadbalance/leader
2022-09-27T22:21:13,428 - INFO  - [metadata-store-12319-1:LeaderElectionImpl@183] - Acquired leadership on /loadbalance/leader
2022-09-27T22:21:13,428 - INFO  - [metadata-store-12319-1:PulsarService@1067] - This broker was elected leader
2022-09-27T22:21:13,428 - WARN  - [metadata-store-12319-1:LeaderElectionImpl@310] - Leadership released for /loadbalance/leader
2022-09-27T22:21:13,433 - INFO  - [metadata-store-12286-1:LeaderElectionImpl@183] - Acquired leadership on /loadbalance/leader
2022-09-27T22:21:13,433 - INFO  - [metadata-store-12286-1:PulsarService@1067] - This broker was elected leader
2022-09-27T22:21:13,433 - WARN  - [metadata-store-12286-1:LeaderElectionImpl@310] - Leadership released for /loadbalance/leader
2022-09-27T22:21:13,434 - INFO  - [TestNG-method=testModularLoadManagerSplitBundle-1:OwnershipCache@193] - Trying to acquire ownership of pulsar/test/localhost:62802/0x00000000_0xffffffff
2022-09-27T22:21:13,438 - INFO  - [metadata-store-12319-1:LeaderElectionImpl@183] - Acquired leadership on /loadbalance/leader
2022-09-27T22:21:13,438 - INFO  - [metadata-store-12319-1:PulsarService@1067] - This broker was elected leader
2022-09-27T22:21:13,438 - WARN  - [metadata-store-12319-1:LeaderElectionImpl@310] - Leadership released for /loadbalance/leader

....
```

#### Location problem
These test uses `setup()` and `PulsarService pulsar_new = startBroker(conf_new)` to create two brokers, but the both brokers use the same `zkc`, so both brokers will share the `session` of `ZK`, which will lead to the possibility that the two brokers will constantly elect for the leader. see code: 

https://github.com/apache/pulsar/blob/b89c1451551a6bbe681465726906a2e61c9d8a69/pulsar-metadata/src/main/java/org/apache/pulsar/metadata/coordination/impl/LeaderElectionImpl.java#L143-L147

#### Impact
- `BrokerServiceLookupTest`

```
constantly elect for the leader makes instability of bundle assign
error occurs
```
- `AdminApi2Test.cleanup`

```
close PulsarService
close LeaderElectionService
delete zk-node `/rebalance/leader`, but this node has concurrently deleted by elect-thread
error occurs
```

### Modifications

- Make two brokers use different `session`

### Documentation

- [x] `doc-not-needed` 

### Matching PR in forked repository

PR in forked repository: 

- https://github.com/poorbarcode/pulsar/pull/15

### CODE REVIEW

**Comment by heesung-sn** [comment]
LGTM.

**Comment by poorbarcode** [comment]
Hi @lhotari @nicoloboschi 

Could you take a look?  Thanks

**Review by gaoran10 — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
Cool!

**Review by liangyepianzhou — APPROVED** [review]
Good work!

**Review by eolivelli — APPROVED** [review]

**Comment by poorbarcode** [comment]
rebase master

**Review by codelipenghui — APPROVED** [review]
👍🏻

**Review by mattisonchao — APPROVED** [review]
Nice work!

**Comment by poorbarcode** [comment]
rebase master

**Review by Technoboy- — APPROVED** [review]

**Comment by tisonkun** [comment]
@poorbarcode you may change the title to `[fix][test] flaky AdminApi2Test.cleanup`

**Comment by poorbarcode** [comment]
Hi @tisonkun 

Got it! Thanks

**Comment by poorbarcode** [comment]
Can this PR merge? (^_^)

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApi2Test.java [hunk 1]
```diff
@@ -147,6 +152,20 @@ protected ServiceConfiguration getDefaultConf() {
         return conf;
     }
 
+    @Override
+    protected MetadataStoreExtended createConfigurationMetadataStore() throws MetadataStoreException {
+        // use MockZooKeeperSession to provide a unique session id for each instance
+        return new ZKMetadataStore(MockZooKeeperSession.newInstance(mockZooKeeperGlobal), MetadataStoreConfig.builder()
```
Comment (nicoloboschi) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApi2Test.java:158]: it shouldn't be better to make this the default behavior in `MockedPulsarServiceBaseTest ` ?

Comment (poorbarcode) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApi2Test.java]: Good idea, already changed

**Pull request merged**